### PR TITLE
	RavenDB-6873 Move the parsing of multipart commands to a separate method

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/BatchHandler.cs [hunk 1]
```diff
@@ -27,14 +27,25 @@ public class BatchHandler : DatabaseRequestHandler
...
+                if (contentType == null ||
+                    contentType.StartsWith("application/json", StringComparison.OrdinalIgnoreCase))
+                {
+                    command.ParsedCommands = await BatchRequestParser.ParseAsync(context, RequestBodyStream(), Database.Patcher);
+                }
+                else if (contentType.StartsWith("multipart/mixed", StringComparison.OrdinalIgnoreCase))
+                {
+                    await ParseMultipart(context, command);
+                }
+
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/BatchHandler.cs:42]: Else, what happens here?

Comment (fitzchak) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:42]: We parse the multipart, which will modify the command?

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:42]: If the content type you "hi/there", what will happen?

Comment (fitzchak) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:42]: Fixed.

**Pull request merged**